crates/fuel-core/src/executor.rs

```diff
@@ -1821,6 +1837,39 @@ mod tests {
             }
         }
 
+        #[test]
+        fn skip_coinbase_during_dry_run() {
+            let price = 1;
+            let limit = 0;
+            let gas_price_factor = 1;
+            let script = TxBuilder::new(2322u64)
+                .gas_limit(limit)
+                // Set a price for the test
+                .gas_price(price)
+                .coin_input(AssetId::BASE, 10000)
+                .change_output(AssetId::BASE)
+                .build()
+                .transaction()
+                .clone();
+
+            let mut config = Config::local_node();
+            let recipient = [1u8; 32].into();
+            config.block_producer.coinbase_recipient = recipient;
+
+            config.chain_conf.transaction_parameters.gas_price_factor = gas_price_factor;
+
+            let producer = Executor::test(Default::default(), config);
+
+            let mut block = Block::default();
+            *block.transactions_mut() = vec![script.into()];
+
+            let ExecutionResult { block, .. } = producer
+                .execute_and_commit(ExecutionBlock::DryRun(block.into()))
```

Comment:
should we add any special logic here to ensure that dry run's won't actually commit? Or even better, enforce it at a type level?

Reply:
We use `execute_and_commit` only for tests, so I marked it with the corresponding compilation feature. Inside the method will panic if someone passes the `ExecutionBlock ::DryRun`